### PR TITLE
cleanup x2j

### PR DESCRIPTION
**Pull request by cdietrich**
cleanup x2j
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/parser/AntlrGrammarComparator.java [hunk 1]
```diff
@@ -8,299 +8,263 @@
...
+			builder.append("|");
+			builder.append(QUOTED_TOKENS, "");
+			builder.append("|[^\\s\'\"");
+			builder.append(Joiner.on("").join(SINGLE_CHAR_TOKENS), "");
+			builder.append("]+");
+			return builder.toString();
+		}
+	}.apply();
+
+	private static String NEWLINE = "\\r?\\n|\\r";
```
Comment (tivervac) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/parser/AntlrGrammarComparator.java:130]: Not static final on these fields?

File: org.eclipse.xtext/src/org/eclipse/xtext/generator/trace/node/TracedAccessorsProcessor.java [hunk 2]
```diff
@@ -26,19 +26,19 @@
...
 import org.eclipse.xtext.xbase.lib.ListExtensions;
 import org.eclipse.xtext.xbase.lib.StringExtensions;
 
 import com.google.common.collect.FluentIterable;
 import com.google.common.collect.Iterables;
+import com.google.common.collect.Sets;
 
 public class TracedAccessorsProcessor extends AbstractClassProcessor {
 
 	private static final Set<String> TYPES_WITH_GOOD_TO_STRING = Collections
```
Comment (szarnekow) [inline org.eclipse.xtext/src/org/eclipse/xtext/generator/trace/node/TracedAccessorsProcessor.java:40]: Java11 -> Set.of(..)?

File: org.eclipse.xtext/src/org/eclipse/xtext/tasks/PreferenceTaskTagProvider.java [hunk 3]
```diff
@@ -36,7 +36,7 @@
 	public static final PreferenceKey CASE_SENSITIVE_KEY = new PreferenceKey("task.caseSensitive", "true");
 
 	public static final List<PreferenceKey> KEYS = Collections.<PreferenceKey>unmodifiableList(
```
Comment (szarnekow) [inline org.eclipse.xtext/src/org/eclipse/xtext/tasks/PreferenceTaskTagProvider.java:38]: List.of

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/parser/AntlrGrammarComparator.java (hunk 1 above)
Comment (tivervac) [reply org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/parser/AntlrGrammarComparator.java:130]: Also, can't you just use `\R` here?

File: org.eclipse.xtext/src/org/eclipse/xtext/generator/trace/node/TracedAccessorsProcessor.java (hunk 2 above)
Comment (cdietrich) [reply org.eclipse.xtext/src/org/eclipse/xtext/generator/trace/node/TracedAccessorsProcessor.java:40]: no java 11 yet

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/parser/AntlrGrammarComparator.java [hunk 4]
```diff
@@ -8,299 +8,263 @@
...
+	 * (key) and the referenced grammar (value) for logging purposes
+	 */
+	public ErrorContext compareGrammars(CharSequence grammar, CharSequence grammarReference,
+			IErrorHandler errorHandler) {
+		if (errorContext == null) {
+			errorContext = new ErrorContext();
+		}
+		Matcher compoundMatcher = compoundPattern.matcher(grammar);
+		Matcher compoundMatcherReference = compoundPattern.matcher(grammarReference);
+		boolean continue_ = true;
```
Comment (tivervac) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/parser/AntlrGrammarComparator.java:199]: Hmm, underscore in names is very weird imo, maybe `cont` is better?

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/parser/AntlrGrammarComparator.java [hunk 5]
```diff
@@ -8,299 +8,263 @@
...
+			}
+			if (!Objects.equal(matchReference, match)) {
+				errorHandler.handleMismatch(match, matchReference, errorContext);
+			}
+		}
+		return errorContext;
+	}
+
+	/**
+	 * continue matching until the next token is found, count newlines while
```
Comment (tivervac) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/parser/AntlrGrammarComparator.java:228]: s/continue/Continue/

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/parser/AntlrGrammarComparator.java [hunk 6]
```diff
@@ -8,299 +8,263 @@
...
+						}
+					}
+				}
+			}
+		}
+		return false;
+	}
+
+	private void handleInvalidGrammarFile(IErrorHandler errorHandler, MatchState state) {
+		if (state == errorContext.testedGrammar) {
```
Comment (tivervac) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/parser/AntlrGrammarComparator.java:264]: Not .equals?

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/parser/AntlrGrammarComparator.java [hunk 7]
```diff
@@ -8,299 +8,263 @@
...
+
+	private static String SL_COMMENT = "//[^\\r\\n]*";
+
+	private static String ML_COMMENT = new Function0<String>() {
+		public String apply() {
+			return "/\\*(\\*[^/]|[^\\*])*\\*/";
+		}
+	}.apply();
+
+	private Pattern p_slComment = Pattern.compile(SL_COMMENT);
```
Comment (tivervac) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/parser/AntlrGrammarComparator.java:142]: Why no camelcase on these vars?

Comment (cdietrich) [reply org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/parser/AntlrGrammarComparator.java]: didnt want to change these or var names

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/parser/AntlrGrammarComparator.java (hunk 1 above)
Comment (cdietrich) [reply org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/parser/AntlrGrammarComparator.java]: finals change was unintended

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/parser/AntlrGrammarComparator.java (hunk 6 above)
Comment (cdietrich) [reply org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/parser/AntlrGrammarComparator.java:264]: didnt want to change any semantics

**Review by szarnekow — APPROVED** [review]

**Pull request merged**